core/blockchain.go

```diff
@@ -880,6 +883,11 @@ func (bc *BlockChain) Stop() {
 		if size, _ := triedb.Size(); size != 0 {
 			utils.Logger().Error().Msg("Dangling trie nodes after full cleanup")
 		}
+	} else {
+		triedb := bc.stateCache.TrieDB()
+		for !bc.trieRef.Empty() {
+			triedb.Dereference(bc.trieRef.PopItem().(common.Hash))
```

Comment:
Why do garbage collect here? This part of the code is executed before the node is shut down. No need to garbage collect any memory right?

Reply:
This part of code is idea from here(non-archival node in Stop() ) https://github.com/harmony-one/harmony/blob/d50073f49c7d1210c5502ca2c4de11d3f8959ffa/core/blockchain.go#L880-L882 , Does this mean that there's actually no need to do garbage collection here either, and it's also redundant code?

Reply:
It’s necessary when the node still runs. But when we exit the node, all memory will be gone. So no need to worry about gc